shaders/src/common_lighting.fs

```diff
@@ -71,6 +75,64 @@ float computeMicroShadowing(float NoL, float visibility) {
     return microShadow * microShadow;
 }
 
+/**
+ * IBL utilities
+ */
+
+vec2 IntersectAABB(vec3 rayOrigin, vec3 rayDir, vec3 boxMin, vec3 boxMax) {
+    vec3 tMin = (boxMin - rayOrigin) / rayDir;
+    vec3 tMax = (boxMax - rayOrigin) / rayDir;
+    vec3 t1 = min(tMin, tMax);
+    vec3 t2 = max(tMin, tMax);
+    float tNear = max3(t1);
+    float tFar = min3(t2);
+    return vec2(tNear, tFar);
+}
+
+// Assume: a <= b
+float GetSmallestPositive(float a, float b) {
+    return a >= 0.0 ? a : b;
+}
+
+/**
+ * This function returns an IBL lookup direction, taking into account the current IBL type (e.g. infinite spherical, 
+ * finite/local sphere/box), an initial intended lookup direction (baseDir) and the particular normal we compute
+ * reflections against (e.g. either the interpolated surface or clearcoat normal).
+ */
+vec3 GetAdjustedReflectedDirection(const vec3 baseDir, const vec3 normal) {
+    vec3 defaultReflected = reflect(-baseDir, normal);    
+
+    if (frameUniforms.iblTechnique == IBL_TECHNIQUE_INFINITE) return defaultReflected;
+
+    // intersect the ray rayPos + t * rayDir with the finite geometry; done in the coordinate system of the finite geometry
+    vec3 rayPos = getWorldPosition() + getWorldOffset() - frameUniforms.iblCenter;
+    vec3 rayDir = defaultReflected;
+
+    vec3  r  = vec3(0.0); // resulting direction
+    float t0 = -1.0f;     // intersection parameter between ray and finite IBL geometry
+    
+    if (frameUniforms.iblTechnique == IBL_TECHNIQUE_FINITE_SPHERE) {
+        // Normalize sphere-space by scaling down positions by radius. We don't scale down ray direction to preserve 
+        // the convenient A = 1 in the quadratic formula. iblHalfExtents.y contains the reciprocal of the IBL sphere radius.
+        vec3 rayPosNormalized = rayPos * frameUniforms.iblHalfExtents.y;
+
+        float B = 2.0 * dot(rayPosNormalized, rayDir);
+        float C = dot(rayPosNormalized, rayPosNormalized) - 1.0; // 1.0 = r^2, as we are in normalized space
+
+        t0 = 0.5 * (-B + sqrt(B*B - 4.0 * C));
+        t0 *= frameUniforms.iblHalfExtents.x;
+    }
```

Comment:
} and else on the same line

Reply:
Fixed in 6dc6210f556cee2f8ce36cfd404588c5ddb3fa2d